test/inductor/test_torchinductor.py

```diff
@@ -4924,6 +4924,87 @@ def fn(a: torch.Tensor) -> torch.Tensor:
             )
             self.assertEqual(arguments_that_are_divisible_by_16_in_kernel1, (0, 1))
 
+    class CudaGraphTests(TestCase):
+        def tearDown(self):
+            torch._inductor.compile_fx.tape_manager_container.tape_manager = None
```

Comment:
can we avoid having global state (or, have a non-global state api that the unit tests can use?)

Reply:
Talked with jason abt this a bit. What makes this difficult is that the compiled function is installed on the bytecode. So if you have something like: 

```
import torch
import torch._dynamo
import torch._inductor
import refcycle
import weakref

foo_dies = False

def test_closure():
    @torch._dynamo.optimize()
    def foo(x):
        return x + 1 + 2

    def finalize():
        global foo_dies
        foo_dies = True

    weakref.finalize(foo, finalize)

    return foo(torch.rand([4], device="cuda"))

out = test_closure()
snapshot = refcycle.snapshot()
```

While the `foo` will get deallocated, the compiled function that is installed on foo's bytecode will not. The second time you invoke `test_closure`, the function is not recompiled.

Could we call `torch._dynamo.reset` before after invoking all pytorch tests ?

Reply:
I added a test where we are showing the correct behavior wrt/ `reset()`.

Reply:
I once tried calling torch._dynamo.reset() before all dynamo unit tests and it made the test suite run irreparably slowly. But yes, resetting seems like the right thing to do for the *dynamo* side integration... but pursuant to my point about this being a valid eager mode only interface, I think you should probably consider doing a bunch of tests without dynamo at all.